internal/core/api.rs

```diff
@@ -421,7 +422,45 @@ impl Window {
     /// Any position fields in the event must be in the logical pixel coordinate system relative to
     /// the top left corner of the window.
     pub fn dispatch_event(&self, event: WindowEvent) {
-        self.0.process_mouse_input(event.into())
+        match event {
+            WindowEvent::PointerPressed { position, button } => {
+                self.0.process_mouse_input(MouseEvent::Pressed {
+                    position: position.to_euclid().cast(),
+                    button,
+                });
+            }
+            WindowEvent::PointerReleased { position, button } => {
+                self.0.process_mouse_input(MouseEvent::Released {
+                    position: position.to_euclid().cast(),
+                    button,
+                });
+            }
+            WindowEvent::PointerMoved { position } => {
+                self.0.process_mouse_input(MouseEvent::Moved {
+                    position: position.to_euclid().cast(),
+                });
+            }
+            WindowEvent::PointerScrolled { position, delta_x, delta_y } => {
+                self.0.process_mouse_input(MouseEvent::Wheel {
+                    position: position.to_euclid().cast(),
+                    delta_x,
+                    delta_y,
+                });
+            }
+            WindowEvent::PointerExited => self.0.process_mouse_input(MouseEvent::Exit),
+            WindowEvent::KeyPressed { modifiers, text } => self.0.process_key_input(&KeyEvent {
+                modifiers,
+                text: SharedString::from(text.encode_utf8(&mut [0; 6]) as &str),
```

Comment:
unrelated, but we probably should implement `From<char> for SharedString`

Reply:
done